host/class/uvc/usb_host_uvc/uvc_host.c

```diff
@@ -48,16 +48,93 @@ void bulk_transfer_callback(usb_transfer_t *transfer);
+    int iface_offset = 0;
+    bool is_uvc_interface = false;
+
+    // Get first Interface descriptor
+    iface_desc = GET_NEXT_INTERFACE_DESC(config_desc, total_length, iface_offset);
+    // Check every uac stream interface
+    while (iface_desc != NULL) {
+        if (iface_desc->bInterfaceClass == USB_CLASS_VIDEO && iface_desc->bInterfaceSubClass == UVC_SC_VIDEOSTREAMING) {
+            // notify user about the connected Interfaces
+            is_uvc_interface = true;
+
+            if (p_uvc_host_driver->user_cb) {
+                const uvc_host_driver_event_data_t conn_event = {
+                    .type = UVC_HOST_DRIVER_EVENT_DEVICE_CONNECTED,
+                    .device_connected.dev_addr = addr,
+                    .device_connected.iface_num = iface_desc->bInterfaceNumber,
+                };
+                p_uvc_host_driver->user_cb(&conn_event, p_uvc_host_driver->user_ctx);
+            }
+
+            const usb_intf_desc_t *iface_alt_desc = GET_NEXT_INTERFACE_DESC(iface_desc, total_length, iface_offset);
+            // Skip all alternate settings belonging to the current interface
+            while (iface_alt_desc != NULL) {
+                // Check if the alternate setting is for the same interface
+                if (iface_alt_desc->bInterfaceNumber != iface_desc->bInterfaceNumber) {
+                    break;
+                }
+                iface_alt_desc = GET_NEXT_INTERFACE_DESC(iface_alt_desc, total_length, iface_offset);
+            }
+            iface_desc = iface_alt_desc;
+            continue;
+        }
+        iface_desc = GET_NEXT_INTERFACE_DESC(iface_desc, total_length, iface_offset);
+    }
+
+    return is_uvc_interface ? ESP_OK : ESP_ERR_NOT_FOUND;
+}
+
+/**
+ * @brief Handler for USB device connected event
+ *
+ * @param[in] addr   USB device physical address
+ * @return esp_err_t
+ */
+static esp_err_t uvc_host_device_connected(uint8_t addr)
+{
+    bool is_uvc_device = false;
+    usb_device_handle_t dev_hdl;
+    const usb_config_desc_t *config_desc = NULL;
+
+    if (usb_host_device_open(p_uvc_host_driver->usb_client_hdl, addr, &dev_hdl) == ESP_OK) {
+        if (usb_host_get_active_config_descriptor(dev_hdl, &config_desc) == ESP_OK) {
+            is_uvc_device = uvc_desc_if_uvc_device(config_desc);
+        }
+        ESP_RETURN_ON_ERROR(usb_host_device_close(p_uvc_host_driver->usb_client_hdl, dev_hdl), TAG, "Unable to close USB device");
+    }
+
+    // Create UAC interfaces list in RAM, connected to the particular USB dev
+    if (is_uvc_device) {
+        // TODO: add config to control it
+        // usb_print_config_descriptor(config_desc, &uvc_print_desc);
+        // Create Interfaces list for a possibility to claim Interface
+        ESP_RETURN_ON_ERROR(uvc_host_interface_check(addr, config_desc), TAG, "uvc stream interface not found");
+    } else {
+        ESP_LOGW(TAG, "USB device with addr(%d) is not UVC device", addr);
+    }
```

Comment:
If we do not pass `iface_num` to the user, this whole part can be removed, simplifying the code a lot. Do we really need it?

Reply:
Some devices have multiple VS interfaces. I would like to notify the user about all the VS interfaces sequentially upon insertion, so that the user can open the interface they wish to use and retrieve all the supported resolutions for that VS interface.

![image](https://github.com/user-attachments/assets/f41c606b-f727-4189-a4ce-1edc07fe6a41)